geode-core/src/main/java/org/apache/geode/management/internal/ContextAwareSSLRMIClientSocketFactory.java

```diff
@@ -44,18 +50,26 @@
   @Override
   public Socket createSocket(String host, int port) throws IOException {
     SocketCreator socketCreator;
+
     try {
       socketCreator =
           SocketCreatorFactory.getSocketCreatorForComponent(SecurableCommunicationChannel.JMX);
+      return socketCreator.connectForClient(host, port, 0);
     } catch (Exception exception) {
-      /*
-       * In the context of a gfsh VM the javax.net.ssl properties are used to configure SSL
-       * appropriately - see the constructor for JMXOperationInvoker.
-       */
-      return defaultFactory.createSocket(host, port);
+      try {
```

Comment:
Like the straight forward approach, is there a specific Exception type we could catch here for this case?

Reply:
Hey @echobravopapa,
We could catch `GemFireConfigException` (it extends `RuntimeException`), but I wanted to  prevent accidentally breaking backward compatibility... in previous versions we always ended up using the default `SslRMIClientSocketFactory` if any `Exception` was thrown when using our own socket factory, so I just wanted to respect that in my changes. Let me know if you think otherwise.

Reply:
that's a good point; can we verify if this would be an issue?